src/QMCDrivers/MCPopulation.cpp

```diff
@@ -64,18 +64,36 @@ void MCPopulation::createWalkers(IndexType num_walkers, const WalkerConfiguratio
 
   outputManager.pause();
 
-  //this part is time consuming, it must be threaded and calls should be thread-safe.
-#pragma omp parallel for
+  // nextWalkerID is not thread safe so we need to get our walker id's here
+  // before we enter a parallel section.
+  std::vector<long> walker_ids(num_walkers_plus_reserve, -1);
+  for (size_t iw = 0; iw < num_walkers; iw++)
+    walker_ids[iw] = nextWalkerID();
```

Comment:
`nextWalkerID` is just horrible.
Why not having
```
computeWalkerID(iw) const
{
  return iw * num_ranks_ + rank_ + 1;
}
```
and get rid of `num_walkers_created_` state

Reply:
that won't work as spawns can occur later.  There needs to be state to not reuse walker ids which will break the whole intent which is for each walker to have an unique id through its life.